samples/ChatRoom/ChatRoom.Common/ChatMsg.cs

```diff
@@ -1,9 +1,13 @@
 namespace ChatRoom;
 
-[Serializable]
-public record class ChatMsg(string? Author, string Text)
+[GenerateSerializer]
+public record class ChatMsg(
+    string? Author,
+    string Text)
 {
+    [Id(2)]
```

Comment:
These can start at zero without colliding with the constructor properties because .
I'm not pleased with how this serializer behavior ended up: we should have kept constructor properties entirely separate from body properties for all `record` types. The way it is, there's room for mistakes and more fragile version tolerance.

Reply:
Updated this to be `0,` and `1`.